pypeit/wavetilts.py

```diff
@@ -749,3 +909,85 @@ def __repr__(self):
         txt += '>'
         return txt
 
+
+def show_tilts_mpl(tilt_img, tilt_traces, slits=None, show_traces=False, cut=None):
+    """Show the TiltImage with the traced and 2D fitted tilts overlaid
+
+    Args:
+        tilt_img (`np.ndarray`_):
+            TiltImage
+        tilt_traces (`astropy.table.Table`_):
+            Table containing the traced and fitted tilts.
+            See :func:`make_tbl_tilt_traces` for information on the table columns.
+        slits (`slittrace.SlitTraceSet`_, optional):
+            SlitTraceSet object
+        show_traces (bool, optional):
+            Show the traced tilts
+        cut (tuple, optional):
+            Lower and upper levels cut for the image display
+    """
+
+    if tilt_traces is None:
+        return msgs.error('No tilts have been traced or fitted')
+
+    if cut is None:
+        cut = utils.growth_lim(tilt_img, 0.98, fac=1)
+
+    w, h = plt.figaspect(1)
+    fig = plt.figure(figsize=(1.5 * w, 1.5 * h))
+
+    plt.imshow(tilt_img, origin='lower', interpolation='nearest', aspect='auto', cmap='gray',
+               vmin=cut[0], vmax=cut[1], zorder=0)
+
+    # traced tilts
+    if show_traces:
+        if 'goodpix_tilt' in tilt_traces.keys() and tilt_traces['goodpix_tilt'][0].size > 0:
+            plt.scatter(tilt_traces['goodpix_spat'][0], tilt_traces['goodpix_tilt'][0], color='cyan',
+                        edgecolors=None, marker='s', s=10, alpha=0.3, lw=0, label='Good pixel', zorder=1)
+    # masked tilts
+    if 'badpix_tilt' in tilt_traces.keys() and tilt_traces['badpix_tilt'][0].size > 0:
+        plt.scatter(tilt_traces['badpix_spat'], tilt_traces['badpix_tilt'], color='red',
+                    edgecolors=None, marker='s', s=10, alpha=0.3, lw=0, label='Masked pixel', zorder=2)
+    # 2D fit tilts
+    # loop over each line so that we can plot lines instead of points
+    if 'good2dfit_lid' in tilt_traces.keys():
+        for iline in np.unique(tilt_traces['good2dfit_lid'][0]):
+            this_line = tilt_traces['good2dfit_lid'][0] == iline
+            if np.any(this_line):
+                good2dfit_spat = tilt_traces['good2dfit_spat'][0][this_line]
+                good2dfit_tilt = tilt_traces['good2dfit_tilt'][0][this_line]
+                plt.plot(good2dfit_spat, good2dfit_tilt, color='blue', lw=1, alpha=0.4, zorder=3)
+
+    # rejected point in 2D fit
+    if 'bad2dfit_tilt' in tilt_traces.keys() and tilt_traces['bad2dfit_tilt'][0].size > 0:
+        plt.scatter(tilt_traces['bad2dfit_spat'][0], tilt_traces['bad2dfit_tilt'][0], color='yellow',
+                    edgecolors=None, marker='s', s=5, alpha=0.3, lw=0, label='Rejected in fit', zorder=4)
+
+    if slits is not None:
```

Comment:
Rather than the dependency on the slit object, why not just add the left, right, mask to the data model. Then this tilts objects is self contained and will plot the slit boundaries that were used to generate the tilts. Otherwise there is lots to keep track of like flexure, etc.